compiler/dialects/stream.py

```diff
@@ -0,0 +1,161 @@
+from collections.abc import Sequence
+from typing import Generic, TypeVar
+
+from xdsl.dialects.builtin import (
+    AffineMapAttr,
+    AnyShapedType,
+    ArrayAttr,
+    ContainerType,
+    StringAttr,
+)
+from xdsl.dialects.utils import AbstractYieldOperation
+from xdsl.ir import (
+    Attribute,
+    Dialect,
+    IsTerminator,
+    ParametrizedAttribute,
+    Region,
+    SSAValue,
+    TypeAttribute,
+)
+from xdsl.irdl import (
+    AttrSizedOperandSegments,
+    IRDLOperation,
+    ParameterDef,
+    irdl_attr_definition,
+    irdl_op_definition,
+    opt_prop_def,
+    prop_def,
+    region_def,
+    var_operand_def,
+    var_result_def,
+)
+
+_StreamTypeElement = TypeVar("_StreamTypeElement", bound=Attribute)
+
```

Comment:
I would add a link here to the other stream dialect implementation and also add the motivation (description of your pull request) in here somewhere.